### PR TITLE
14044 export and import client applications

### PR DESCRIPTION
**Pull request by amartinj**
Closes https://github.com/CartoDB/cartodb/issues/14044

### CODE REVIEW

File: spec/services/carto/user_metadata_export_service_spec.rb [hunk 1]
```diff
@@ -725,12 +785,52 @@ def full_export_import_viewer(path)
...
 
+  let(:full_export_one_zero_five) do
+    user_hash = full_export[:user].except!(:client_application)
+    full_export[:user] = user_hash
+    full_export
+  end
+
   let(:full_export_one_zero_four) do
-    user_hash = full_export[:user].except!(:synchronization_oauths).except!(:connector_configurations)
+    user_hash = full_export_one_zero_five[:user].except!(:synchronization_oauths).except!(:connector_configurations)
```
Comment (AleGoiko) [inline spec/services/carto/user_metadata_export_service_spec.rb]: I think something like  `full_export_one_zero_five[:user].except!([:synchronization_oauths, :connector_configurations])` looks cleaner (i know its old, i just thought it looked weird :P)

File: spec/services/carto/user_metadata_export_service_spec.rb [hunk 2]
```diff
@@ -81,6 +99,7 @@ def destroy_user(user = @user)
       st.destroy
     end
     user.destroy
+    ::User[user.id].before_destroy(skip_table_drop: true)
```
Comment (AleGoiko) [inline spec/services/carto/user_metadata_export_service_spec.rb]: :thinking:

File: spec/services/carto/user_metadata_export_service_spec.rb [hunk 3]
```diff
@@ -63,6 +63,24 @@ def create_user_with_basemaps_assets_visualizations
...
+    sequel_user.client_application.oauth_tokens << ::OauthToken.new(
+      token: "oauth_token",
+      secret: "oauth_secret",
+      callback_url: "http//callback.com",
+      verifier: "v1",
+      scope: nil,
+      client_application_id: sequel_user.client_application.id
+    ).save
+
     @user.reload
```
Comment (AleGoiko) [inline spec/services/carto/user_metadata_export_service_spec.rb]: why reload twice?

File: app/services/carto/user_metadata_export_service.rb [hunk 4]
```diff
@@ -75,6 +75,11 @@ def build_search_tweets_from_hash_export(exported_hash)
     def save_imported_user(user)
       user.save!
       ::User[user.id].after_save
+      user.client_applications.first.access_tokens.each do |t|
```
Comment (AleGoiko) [inline app/services/carto/user_metadata_export_service.rb]: :thinking:

File: spec/services/carto/user_metadata_export_service_spec.rb (hunk 2 above)
Comment (amartinj) [reply spec/services/carto/user_metadata_export_service_spec.rb]: `user` is `Carto::User` and we need to run some sequel callbacks 💩

File: spec/services/carto/user_metadata_export_service_spec.rb (hunk 3 above)
Comment (amartinj) [reply spec/services/carto/user_metadata_export_service_spec.rb]: Again, two orms = 💩. We are updating the database using sequel, so we need `@user` to be aware of those changes.

File: app/services/carto/user_metadata_export_service.rb (hunk 4 above)
Comment (amartinj) [reply app/services/carto/user_metadata_export_service.rb]: Again... Two ORM pain. We need to create the data using AR (so we avoid some callbacks to be run), but we also need to run some of them... again 💩

**Comment by AleGoiko** [comment]
https://media0.giphy.com/media/3o8doT9BL7dgtolp7O/giphy.gif

**Pull request merged**